functional_tests/step_helpers/release_page_helpers.py

```diff
@@ -0,0 +1,272 @@
+    add_description_block(context, index=1)
+
+
+def add_invalid_release_date_text(context: Context):
+    fill_locator(context, RELEASE_DATE_TEXT, "Invalid 5555")
+
+
+def add_invalid_next_release_date_text(context: Context):
+    fill_locator(context, NEXT_RELEASE_DATE_TEXT, "Invalid 5555")
+
+
+def add_next_release_before_release(context: Context):
+    fill_locator(context, RELEASE_DATE, "2025-12-25")
+    fill_locator(context, NEXT_RELEASE_DATE, "2024-12-25")
+
+
+def add_both_next_release_dates(context: Context):
+    fill_locator(context, NEXT_RELEASE_DATE, "2025-12-25")
+    fill_locator(context, NEXT_RELEASE_DATE_TEXT, "December 2024")
+
+
+def expect_related_links(context: Context):
+    expect(context.page.get_by_role("heading", name="You might also be interested")).to_be_visible()
+    expect(context.page.locator("#links").get_by_role("link", name="Home")).to_be_visible()
+
+
+def expect_not_both_dates_error(context: Context):
+    expect(
+        context.page.locator(
+            "#panel-child-content-child-metadata-child-panel1-child-next_release_date-errors"
+        ).get_by_text("Please enter the next release date or the next release text, not both.")
+    ).to_be_visible()
+    expect(
+        context.page.locator(
+            "#panel-child-content-child-metadata-child-panel1-child-next_release_date_text-errors"
+        ).get_by_text("Please enter the next release date or the next release text, not both.")
+    ).to_be_visible()
+
+
+def expect_changes_to_release_date(context: Context):
+    preview_texts: list[str] = [
+        "Previous date",
+        "25 December 2024 9:30am",
+        "Reason for change",
+        "Updated due to data availability",
+    ]
+    expect_text(context, None, preview_texts)
+    expect(context.page.get_by_role("link", name="Changes to this release date")).to_be_visible()
+    expect(context.page.get_by_role("heading", name="Changes to this release date")).to_be_visible()
+
+
+def add_multiple_descriptions(context: Context):
+    add_description_block(context)
+    insert_block(context, block_name="Description", index=2)
+
+
+def add_multiple_tables(context: Context):
+    insert_block(context, block_name="Basic table")
+    insert_block(context, block_name="Basic table", index=1)
+
+
+def add_table_no_header(context: Context):
+    add_basic_table(context, header=False)
+
+
+def add_empty_table(context: Context):
+    add_basic_table(context, data=False)
+
+
+def add_release_date_change_no_log(context: Context):
+    context.page.get_by_role("textbox", name="Release date*").fill("2025-01-25")
+
+
+def add_another_release_date_change(context: Context):
+    section = context.page.locator("#panel-child-content-changes_to_release_date-section")
+    section.get_by_role("button", name="Insert a block").nth(1).click()
+    section.get_by_label("Reason for change*").nth(1).fill("New update to release schedule")
+
+
+# Dispatcher mapping
+FEATURE_ACTIONS: dict[str, Callable[[Context], None]] = {
```

Comment:
I'd move this to the top, before the functions as it is a bit mixed in and lost here